web-common/src/features/sources/modal/submitAddDataForm.ts

```diff
@@ -193,6 +191,25 @@ export async function submitAddConnectorForm(
         details: result.details,
       };
     }
+  } else {
+    // Poll for the non-OLAP connectors resource status
+    // If the resource status is not reconciled, rollback the changes
+    const result = await pollConnectorResource(
+      instanceId,
+      connector.name as string,
+    );
```

Comment:
It looks like this `pollConnectorResource` function includes logic that's redundant to what exists above in this `submitAddConnectorForm` function. 

For example:
- `runtimeServicePutFileAndWaitForReconciliation` waits for reconciliation of the resource, so we shouldn't need the polling logic in `pollConnectorResource`, right?
- `fileArtifacts.checkFileErrors` checks for resource reconciliation errors, so we additionally shouldn't need to check again in `pollConnectorResource`, right?

Reply:
In general, as much as possible, we should use the same code path for all connectors (both non-OLAP and OLAP).

Reply:
As discussed in our check-in, for non-OLAP connectors, I don't think it's necessary to do a "test connection" client-side. Per [Naman's PR](https://github.com/rilldata/rill/pull/7426), the runtime's reconciliation process should be testing the connection server-side. You can look at that PR to see which connectors are covered by his work. For example, here's the Athena logic:

<img width="1740" height="479" alt="image" src="https://github.com/user-attachments/assets/f15d04fc-9f2b-4f83-9453-65b0c97e0fcb" />

Reply:
Addressing

Reply:
Reverted the calling `ListDatabaseSchemas` API changes. I think we should still [use a polling](https://linear.app/rilldata/issue/APP-23/add-test-connection-feature-to-all-connectors-ui#comment-df1c7978) to check for the server's reconcile status. I tried the approach by just relying on `runtimeServicePutFileAndWaitForReconciliation`. It will continue to process the resource in the server side but it won't reflect the reconcile status changes to the UI.

Reply:
> Non-OLAP Connectors:
> 1. Create connector file with runtimeServicePutConnectorFileAndWaitForReconciliation
> 2. Wait for reconciliation + connection testing during reconciliation
> 3. Create .env file with runtimeServicePutFileAndWaitForReconciliation

As mentioned in my latest review, we'll need to create the `.env` file _before_ testing the connection.

Reply:
In general, the `submitAddConnectorForm` currently requires a very careful read to understand the difference between OLAP connectors and non-OLAP connectors. To strive toward _one_ codepath, we could request that Platform tests the OLAP connectors server-side during reconciliation – just like they do for the non-OLAP connectors.